### PR TITLE
Allow converting Task Scratch Org into Dev Org.

### PR DESCRIPTION
**Pull request by jgerigmeyer**
## Cute animal pic
_Because everyone likes pictures of [animals](https://unsplash.com/t/animals)._

## Link to Trello card (if applicable)
_If this PR relates to a Trello card, don't forget to reference this PR on the card as well._

https://trello.com/c/1wrCiCiM/233-i-want-to-convert-a-task-scratch-org-into-a-dev-org-for-that-task

## Description
_The commit messages say what you did; this should explain why and/or how._
- [x] Description is in Trello card

## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally (including the pages/URLs/views/states to review)._

- Create a **Task** Scratch Org
- Make changes in the org
- Click "contribute" btn on org
- Select either "Convert Scratch Org into Dev Org on a new Task" or "Convert Scratch Org into Dev Org for this Task" (disabled if there is already a Dev Org on the task)
- Click "Contribute" btn
- If you selected "new task", fill out form, then submit and see the new task with your org as the Dev Org (and you assigned as the Dev), then the "retrieve changes" modal opens
- If you selected "this task", see your org now as the Dev Org (and you assigned as the Dev, if you weren't before), then the "retrieve changes" modal opens

## Show me
_Provide screenshots/animated gifs/videos if necessary._

Screen Shot 2021-06-04 at 3 51 30 PM

Screen Shot 2021-06-04 at 3 51 36 PM

Screen Shot 2021-06-04 at 3 52 40 PM

### CODE REVIEW

**Comment by jgerigmeyer** [comment]
While I back-fill test coverage, I'm requesting code and/or functional review from some combination of @stacyk, @jerivas and @dvdherron. I don't think there are any styles or backend changes required.

The goal is to get this reviewed and merged on Monday.

**Comment by jgerigmeyer** [comment]
[I want to convert a task scratch org into a dev org for that task.](https://trello.com/c/1wrCiCiM/233-i-want-to-convert-a-task-scratch-org-into-a-dev-org-for-that-task)

**Review by jerivas — APPROVED** [review]
Looking (and working) good!

File: locales_dev/en/translation.json [hunk 1]
```diff
@@ -374,6 +377,8 @@
   "confirmRemoveCollaboratorsMessage": "The following user is being removed from this epic, but is already assigned to at least one task. Removing this user will not remove them from any assigned tasks. Are you sure you want to remove this user from the epic?",
   "confirmRemoveCollaboratorsMessage_plural": "The following user is being removed from this epic, but is already assigned to at least one task. Removing this user will not remove them from any assigned tasks. Are you sure you want to remove this user from the epic?",
   "contributeWorkFromEpic": "<0><0>To contribute the work you’ve done in your scratch org,</0> you’ll start by creating a new task.</0><1>Your scratch org will become the Dev Org for the newly created task.</1>",
```
Comment (dvdherron) [inline locales_dev/en/translation.json:379]: Is there a distinction between lowercase `scratch org` and uppercase `Scratch Org`?

**Review by dvdherron — APPROVED** [review]
LGTM

File: locales_dev/en/translation.json (hunk 1 above)
Comment (jgerigmeyer) [reply locales_dev/en/translation.json:379]: That's a great question. I've seen both `Scratch Org` and `scratch org` (similarly both `project` and `Project` etc) in the mockups and the live site, and I'm not 100% sure if we have a clear rule. @SondraE or @stacyk do you want to weigh in here?

It'd probably be a good idea to create a new story to track this, where we can develop and document a clearer style guide.

Comment (stacyk) [reply locales_dev/en/translation.json:379]: I believe we were leaning towards having all of the Objects capitalized... but I think we were converting those as we moved along and didn't always catch all of the terms retroactively. I know Projects, Tasks, Epics, Collaborators were the things we started to capitalize, but maybe we could develop more concrete guidelines and document them somewhere. @SondraE what is your understanding?

Comment (jgerigmeyer) [reply locales_dev/en/translation.json:379]: That makes sense to me. I'll make a new card to more carefully review and update the current usage. 👍 

https://trello.com/c/J04dg5Yi/293-i-want-consistent-capitalization-of-core-objects-terms-eg-project-task-scratch-org

Comment (SondraE) [reply locales_dev/en/translation.json:379]: Yep. As I was writing text for the self-guided tour, I was capitalizing Project, Epic, Task, Collaborator, and Org to help me remember what the important objects were. Then I realized capitalizing these words might help users grasp their importance in an application that has a lot of terms to learn.

**Review by stacyk — APPROVED** [review]
This is beautiful. I think this feature (and the Epic one) will really help contributors.

**Pull request merged**